src/components/button/button.tsx

```diff
@@ -1,27 +1,61 @@
 import React from "react";
-import styled from "styled-components";
-import { fontStyle, fontWeight, spacing, fontFamily, fontSize } from "../../styles"
-
-interface Props {
-    color: string;
-    border: string;
-    width: string;
-    height: string;
-    backgroundColor: any;
-    // add props that come into the component
-}
+import styled, {css} from "styled-components";
+import {
+    breakpoint, mediaQuery,
+    pxToRem,
+} from "../../styles";
+type VariantType = "primary" | "secondary" | "pure";
+type ColorType = "default" | "info" | "error" | "warning" | "main"| "logo";
+type Props = {
+    color: ColorType;
+    variant: VariantType;
+    children?: React.ReactNode;
+};
 
 const StyledButton = styled.button`
-  color: ${(props: any) => props.color};
-  border: ${(props: any) => props.border};
-  width: ${(props: any) => props.width};
-  height: ${(props: any) => props.height};
-  background-color: ${(props: any) => props.backgroundColor};
-  margin: ${(props: any) => props.margin};
-`
-
+  color: ${(props: any) => props.variant === "primary" ? "white" : props.variant === "secondary" ? props.theme.notification[props.color] : 'white'};
```

Comment:
instead of `white` we should use the HEX value which is more detailed. If not in the `theme` defined already we should add and use the color from the `theme`object